cms/djangoapps/contentstore/outlines.py

```diff
@@ -0,0 +1,116 @@
+"""
+This is where Studio interacts with the learning_sequences application, which
+is responsible for holding course outline data. Studio _pushes_ that data into
+learning_sequences at publish time.
+"""
+from datetime import timezone
+
+from edx_django_utils.monitoring import function_trace, set_custom_attribute
+
+from openedx.core.djangoapps.content.learning_sequences.api import replace_course_outline
+from openedx.core.djangoapps.content.learning_sequences.data import (
+    CourseLearningSequenceData,
+    CourseOutlineData,
+    CourseSectionData,
+    CourseVisibility,
+    ExamData,
+    VisibilityData,
+)
+from xmodule.modulestore import COURSE_ROOT, LIBRARY_ROOT, ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+
+
+@function_trace('get_outline_from_modulestore')
+def get_outline_from_modulestore(course_key):
+    """
+    Get a learning_sequence.data.CourseOutlineData for a param:course_key
+    """
+    def _remove_version_info(usage_key):
+        """
+        When we ask modulestore for the published branch in the Studio process
+        after catching a publish signal, the items that have been changed will
+        return UsageKeys that have full version information in their attached
+        CourseKeys. This makes them hash and serialize differently. We want to
+        strip this information and have everything use a CourseKey with no
+        version information attached.
+
+        IF we can verify that this property is still true when running in an
+        out-of-process celery task, it might be a path towards optimizing our
+        publish updates to only what has changed–but that carries other risks as
+        well in terms of correctness (e.g. race conditions).
```

Comment:
Yeah. Based on a conversation with Cale, it seems like it's more a side-effect than intentional design–certainly nothing that we have API guarantees about. And I strongly suspect that it doesn't show up when we run celery in a separate process–I just haven't bothered to test it since it's not normal for devstack and there's a lot more work to do in order to make that work properly. I'll try to clarify the comment.

Reply:
Comment looks great :+1: